pkg/controller/service/service_controller.go

```diff
@@ -582,7 +591,45 @@ func nodeSlicesEqualForLB(x, y []*v1.Node) bool {
 	return nodeNames(x).Equal(nodeNames(x))
 }
 
-func getNodeConditionPredicate() corelisters.NodeConditionPredicate {
+func getNodeTaintPredicate() corelisters.NodePredicate {
+	return func(node *v1.Node) bool {
+		// We add the master to the node list, but its unschedulable.  So we use this to filter
+		// the master.
+		if node.Spec.Unschedulable {
+			return false
+		}
+		// If no taint, accept
+		if len(node.Spec.Taints) == 0 {
+			return true
+		}
+		// As of 1.6, we will taint the master, but not necessarily mark it unschedulable.
+		// Recognize nodes labeled as master, and filter them also, as we were doing previously.
+		if _, hasMasterRoleLabel := node.Labels[LabelNodeRoleMaster]; hasMasterRoleLabel {
+			return false
+		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.ServiceNodeExclusion) {
+			if _, hasExcludeBalancerLabel := node.Labels[LabelNodeRoleExcludeBalancer]; hasExcludeBalancerLabel {
+				return false
+			}
+		}
+		// Not Ready, no execute taint
+		notReadyTaintTemplate := &v1.Taint{
```

Comment:
there are two of them: notReady and unreachable. It's a bit missealing in an original code, but Conditions use 3-value logic, so `!= true` == `== false || == unknown`.

Reply:
Finaly dont need to check a taint globally (key + effect) if we only check for the effect